frmts/georaster/georaster_wrapper.cpp

```diff
@@ -539,6 +539,66 @@ GeoRasterWrapper *GeoRasterWrapper::Open(const char *pszStringId, bool bUpdate)
     return poGRW;
 }
 
+bool ValidateInsertExpression(const CPLString &sInsertStatement);
+bool ValidateDescriptionExpression(const CPLString &sInsertStatement);
+
+bool ValidateInsertExpression(const CPLString &sInsertStatement)
+{
+    std::vector<char> vStringExpressions;
+    for (size_t nPos = 0; nPos < sInsertStatement.length(); ++nPos)
+    {
+        // Check if text is inside quotes
+        if (sInsertStatement[nPos] == '\'' &&
+            sInsertStatement[nPos - 1] != '\\')
+        {
+            if (vStringExpressions.empty())
+            {
+                vStringExpressions.push_back('\'');
+            }
+            else if (vStringExpressions.back() == '\'')
+            {
+                vStringExpressions.pop_back();
+            }
+        }
+
+        // Check that, if text ';', '--', '/*' and '*/' exists, is only inside quotes.
+        if (sInsertStatement[nPos] == ';' ||
+            (sInsertStatement[nPos] == '-' && 
+            sInsertStatement[nPos + 1] == '-') ||
```

Comment:
could nPos +1 > length?

Reply:
The loop needs to end in sInsertStatement.length()-1 to avoid the case when the last character is any of those that require to check the next character.